middleware/ledger.js

```diff
@@ -16,23 +16,32 @@ module.exports = async function useLedgerSigner({ useLedgerKey: ledgerKeyPath, n
     const client = await createClient(transport);
 
     let cachedPublicKeys = {};
-    async function getPublicKeyForPath(hdKeyPath) {
+    async function getPublicKeyForPath(hdKeyPath, enableCashing = true) {
         // NOTE: Public key is cached to avoid confirming on Ledger multiple times
-        if (cachedPublicKeys[ledgerKeyPath]) {
+        if (cachedPublicKeys[ledgerKeyPath] && enableCashing) {
             return cachedPublicKeys[hdKeyPath];
         }
-
         console.log('Waiting for confirmation on Ledger...');
-        const rawPublicKey = await client.getPublicKey(hdKeyPath);
+        let rawPublicKey = '';
+        try {
+            rawPublicKey = await client.getPublicKey(false, true);
```

Comment:
nit: very unclear what `false` and `true` use here. Best to pass named variables in such case

Reply:
https://softwareengineering.stackexchange.com/a/161974